libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2412,6 +2412,29 @@ string YulUtilFunctions::conversionFunctionSpecial(Type const& _from, Type const
 			.render();
 		}
 
+		if (_from.category() == Type::Category::Array && _to.category() == Type::Category::Array)
+		{
+			auto const& fromArrayType = dynamic_cast<ArrayType const&>(_from);
+			auto const& toArrayType = dynamic_cast<ArrayType const&>(_to);
+
+			solAssert(!fromArrayType.baseType()->isDynamicallyEncoded(), "");
+			solAssert(fromArrayType.isByteArray() && toArrayType.isByteArray(), "");
+			solAssert(toArrayType.location() == DataLocation::Memory, "");
+			solAssert(fromArrayType.location() == DataLocation::CallData, "");
+			solAssert(toArrayType.isDynamicallySized(), "");
+
+			Whiskers templ(R"(
+				function <functionName>(offset, length) -> converted {
+					converted := <allocateMemoryArray>(length)
+					<copyToMemory>(offset, add(converted, 0x20), length)
```

Comment:
```
	/// @returns a function that copies raw bytes of dynamic length from calldata
	/// or memory to memory.
	/// Pads with zeros and might write more than exactly length.
	std::string copyToMemoryFunction(bool _fromCalldata);
```

Reply:
Ah, I see that it actually is fine - the array allocation function allocates enough.

Reply:
Rounding up does not help, but I think we can assume that we can write past the end here because it is freshly allocated.